### DISCUSSION TITLE
A default value for a Controller component isn't considered valid without manually touching the component via UI

### DISCUSSION THREAD

**Discussion by risanto** [react-hook-form/react-hook-form#5998] [Q&A]
I have an identical issue to this PR: https://github.com/react-hook-form/react-hook-form/issues/857

I already tried hoisting like this but it doesn't do anything.
```jsx
const { handleSubmit, control, formState } = useForm<FormValues>();
const { errors, isValid } = formState;
```
Am I supposed to use isValid anywhere? Because at the moment I'm not using isValid (except on a console.log). All the values are shown correctly in the input components but formState.isValid stays false if I don't change the input manually from the UI.

**Comment by bluebill1049** [comment]
i think you didn't provide `defaultvalues` at useForm

**Reply by risanto** [reply]
Ok solved, thank you so much.

Another question: am I supposed provide defaultValues at both useForm and component if I use Controller?

For example:
```jsx
  const defaultValues = useMemo(() => latestData?.step_2, [latestData]);
  const {
    handleSubmit,
    control,
    formState: { errors }
  } = useForm<FormValues>({ defaultValues: defaultValues });

  function renderInputController(
    name: string,
    placeholder: string
  ) {
    return (
      <Controller
        control={control}
        name={name}
        rules={{
          validate: (val) => {
            if (!val) {
              return 'Kolom wajib diisi';
            }
          }
        }}
        render={({ field: { onChange } }) => (
          <Input
            placeholder={placeholder}
            onChange={onChange}
            error={errors[name]?.message}
            defaultValue={defaultValues?.[name]}
          />
        )}
      />
    );
  }
```
If I don't pass the prop defaultValue, Input stays empty.

**Comment by bluebill1049** [comment]
<img width="943" alt="Screen Shot 2021-07-19 at 3 45 30 pm" src="https://user-images.githubusercontent.com/10513364/126108896-c5cf6e11-a6aa-45d0-8db9-4a4bcb03a362.png">

https://react-hook-form.com/api/usecontroller

follow the doc ^
